docs/data/base/components/number-input/number-input.md

```diff
@@ -169,6 +169,87 @@ In the following snippet, if <kbd class="key">Shift</kbd> is held when clicking
 <NumberInput min={0} step={1} shiftMultiplier={5} />
 ```
 
+### Events
+
+The number input component and hook provides two props–`onChange` and `onInputChange`–that accept event handlers for when the value of the component changes.
+
+#### onChange
+
+`onChange` accepts a custom event handler that is called with two arguments: the underlying event, and the latest "clamped" value.
+
+```ts
+onChange: (
+  event: React.FocusEvent<HTMLInputElement> | React.PointerEvent | React.KeyboardEvent,
+  value: number | undefined,
+) => void;
+```
+
+It's called when the `<input>` element is blurred, or when the stepper buttons are clicked, after the value has been clamped based on the [`min`, `max`](#minimum-and-maximum), or [`step`](#incremental-steps) props.
+
+:::warning
+When using the component, `onChange` can only be passed as a prop on the component—not through `slotProps`.
+:::
+
+```jsx
+// ✅ Works
+<NumberInput
+  onChange={(event, newValue) => console.log(`${event.type} event: the new value is ${newValue}`)}
+/>
+
+// ❌ Doesn't work
+<NumberInput
+  slotProps={{
+    input: {
+      // expects a native input change event handler, newValue is always undefined
+      onChange: (event, newValue) => { ... },
+    },
+  }}
+/>
+```
+
+#### onInputChange
+
+`onInputChange` accepts a native input change handler that's passed to the `<input>` element:
+
+```ts
+onInputChange: React.ChangeEventHandler<HTMLInputElement>;
+```
+
+It's called whenever the value of the textbox changes–for example, on every keystroke typed into it, before clamping is applied.
+
+In other words, it's possible for `event.target.value` to contain out-of-range values or non-numerical characters.
+
+:::warning
```

Comment:
IMO it could be an "info".

I'd also underline that both ways (`onInputChange` and `slotProps.input.onChange`) work the same way.

Reply:
Updated!